src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -1221,13 +1280,23 @@ fn load_stratis_metadata(
             )));
     }
 
-    match get_metadata(&infos) {
-        Ok(opt) => opt
+    match get_metadata(&stratis_infos_ref(infos)) {
+        Ok(opt) => {
+            let (timestamp, mut metadata) = opt
             .ok_or_else(|| {
                 StratisError::Msg(format!(
                     "No metadata found on devices associated with pool UUID {pool_uuid}"
                 ))
-            }),
+            })?;
+            let dev_uuids_and_paths = if remove_cache {
```

Comment:
Minor stylistic point: Our lowest supported Rust is 1.77.0 these days, and that means that we can use ```bool.then()``` here! https://doc.rust-lang.org/std/primitive.bool.html#method.then .